app/assets/scripts/views/ministry.js

```diff
@@ -1,19 +1,131 @@
 'use strict';
 import React from 'react';
+import path from 'path';
+import { connect } from 'react-redux';
+import { get } from 'object-path';
+import Share from '../components/share';
+import Map from '../components/map';
+import ProjectCard from '../components/project-card';
+import HorizontalBarChart from '../components/charts/horizontal-bar';
+import { shortTally, tally, shortText } from '../utils/format';
 
 var Ministry = React.createClass({
   displayName: 'Ministry',
 
+  propTypes: {
+    params: React.PropTypes.object,
+    location: React.PropTypes.object,
+    api: React.PropTypes.object,
+    meta: React.PropTypes.object
+  },
+
   render: function () {
+    const projects = get(this.props, 'api.projects', []);
+    if (projects.length === 0) {
+      return <div></div>; // TODO loading indicator
+    }
+    const basepath = '/' + this.props.meta.lang;
+
+    const ministryName = decodeURIComponent(this.props.params.name);
```

Comment:
Ah, I thought this was in reference to something on the API side. I think we should use the slugify module in `util` for this, you can match it against the slugged names of the dataset, like it's done in `donors.js`.